public/app/plugins/datasource/cloudwatch/metric-math/completion/CompletionItemProvider.ts

```diff
@@ -0,0 +1,116 @@
+
+type CompletionItem = monacoTypes.languages.CompletionItem;
+
+export class MetricMathCompletionItemProvider extends CompletionItemProvider {
+  constructor(datasource: CloudWatchDatasource, templateSrv: TemplateSrv = getTemplateSrv()) {
+    super(datasource, templateSrv);
+    this.getStatementPosition = getStatementPosition;
+    this.getSuggestionKinds = getSuggestionKinds;
+    this.tokenTypes = MetricMathTokenTypes;
+  }
+
+  async getSuggestions(
+    monaco: Monaco,
+    currentToken: LinkedToken | null,
+    suggestionKinds: SuggestionKind[],
+    statementPosition: StatementPosition,
+    position: monacoTypes.IPosition
+  ): Promise<CompletionItem[]> {
+    let suggestions: CompletionItem[] = [];
+    const invalidRangeToken = currentToken?.isWhiteSpace() || currentToken?.isParenthesis();
+    const range =
+      invalidRangeToken || !currentToken?.range ? monaco.Range.fromPositions(position) : currentToken?.range;
+
+    const toCompletionItem = (value: string, rest: Partial<CompletionItem> = {}) => {
+      const item: CompletionItem = {
+        label: value,
+        insertText: value,
+        kind: monaco.languages.CompletionItemKind.Field,
+        range,
+        sortText: CompletionItemPriority.Medium,
+        ...rest,
+      };
+      return item;
+    };
+
+    function addSuggestion(value: string, rest: Partial<CompletionItem> = {}) {
+      suggestions = [...suggestions, toCompletionItem(value, rest)];
+    }
+
+    for (const suggestion of suggestionKinds) {
+      switch (suggestion) {
+        case SuggestionKind.FunctionsWithArguments:
+          METRIC_MATH_FNS.map((f) =>
+            addSuggestion(f, {
+              insertText: f === 'SEARCH' ? `${f}('$0')` : `${f}($0)`,
+              insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet,
+              command: TRIGGER_SUGGEST,
+              kind: monaco.languages.CompletionItemKind.Function,
+            })
+          );
+          break;
+
+        case SuggestionKind.KeywordArguments:
+          METRIC_MATH_KEYWORDS.map((s) =>
+            addSuggestion(s, {
+              insertText: s,
+              command: TRIGGER_SUGGEST,
+              kind: monaco.languages.CompletionItemKind.Keyword,
+              sortText: CompletionItemPriority.MediumHigh,
+            })
+          );
+          break;
+
+        case SuggestionKind.Statistic:
+          METRIC_MATH_STATISTIC_KEYWORD_STRINGS.map((s) =>
+            addSuggestion(s, {
+              insertText: `'${s}', `,
+              command: TRIGGER_SUGGEST,
+            })
+          );
+          break;
+
+        case SuggestionKind.Operators:
+          METRIC_MATH_OPERATORS.map((s) =>
+            addSuggestion(s, {
+              insertText: `${s} `,
+              command: TRIGGER_SUGGEST,
+            })
+          );
+
+        case SuggestionKind.Period:
+          METRIC_MATH_PERIODS.map((s) => addSuggestion(s.toString()));
```

Comment:
It's unbelievable that monaco is not capable of sorting numeric values in order. :| Maybe you can find some way of working around that though. Something like this could work `[{insertText: 60, sortText: 'a'}, ...]`

![Screenshot from 2022-01-26 08-19-57](https://user-images.githubusercontent.com/2388950/151135119-a0df7bd7-ed71-4b56-9941-1a9609d6765b.png)